### ISSUE TITLE
net 8 console app making use of Microsoft.Extensions.* NuGet Package won't deploy without aspnet runtime

### ISSUE THREAD

**Issue by a-teece** [dotnet/runtime#100009]
### Description

If a console (or other Windows Desktop app) makes use of a library that makes use of many of the Microsoft.Extensions.* nuget packages then at runtime the machine must have both the .net Windows Desktop runtime and the Asp Net runtime deployed, even though this application makes no use of Asp.Net.

In my specific example we have a Console and a Windows Service Applications connecting to Office 365 via the Graph API which makes use of the NuGet package "Microsoft.Identity.Web.TokenCache" in order to cache various auth tokens. This package makes use of "Microsoft.Extensions.Caching.Memory" which makes use of "Microsoft.Extensions.DependencyInjection.Abstractions".

When running on a machine with the Windows Desktop runtime pack installed we get exceptions of:
System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.Extensions.DependencyInjection.Abstractions, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. The system cannot find the file specified.

This is solved by deploying the AspNet runtime pack as well as the Windows Desktop - but that seems strange given it isn't an ASPNet app and is extra deployment steps.

### Reproduction Steps

Create a console app project targeting net8.0 and set CopyLocalLockFileAssemblies to true in the csproj file.
Add a NuGet reference to Microsoft.Extensions.Caching.Memory
Build

Expect that the bin folder contains all the dependencies.
Copy the bin folder to a Windows Server with the .net 8.0 desktop pack deployed
Try to run the console app

### Expected behavior

I would expect that when "CopyLocalLockFileAssemblies" these dependencies are in the bin folder unless the application is targeting the Microsoft.AspNetCore.App framework OR include these non-AspNet assemblies in the Windows Desktop pack too.

Simply put C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App\8.0.3 contains a bunch of assemblies that can and are used by non AspNet applications and it is confusing continuing to consider them as AspNet-only dependencies

### Actual behavior

Some dependant assemblies are missing in the bin (output) folder.

Exceptions of:
System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.Extensions.DependencyInjection.Abstractions, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. The system cannot find the file specified.

### Regression?

Yes. This worked correctly in .Net Framework 4.7.2

### Known Workarounds

Deploy the AspNetCore runtime pack to the target machine - even though it is not an AspNet application.

**Comment by huoyaoyuan** [comment]
The dependency injection packages are available as standalone nuget packages. When creating a new console project referencing Microsoft.Identity.Web.TokenCache, I can see Microsoft.Extensions.DependencyInjection.Abstractions.dll in the bin folder.

How is your application deployed? ASP.NET runtime won't be required without FrameworkReference to it. It just happens to carry the missing library.

**Comment by a-teece** [comment]
@huoyaoyuan So I just tried completely blank solution and couldn't replicate until I added "Microsoft.Azure.SignalR.Management" which we are using from our Console/Service app to send messages too users in a SPA.

Once this package is referenced loads of assemblies are missing from the BIN folder even though it is still not an ASP.NET project.

**Comment by a-teece** [comment]
I didn't realise Microsoft.Azure.SignalR.Management was part of the problem and thought it was more generic than that

**Comment by a-teece** [comment]
Digging very deeply through the MSBuild log makes it look like the "UpdateAspNetToFrameworkReference" target is removing at least 1 package reference and is injecting a framework reference of "Microsoft.AspNetCore.App".

I have tried setting "DisableImplicitFrameworkReferences" to true and then explicitly listing my 2 framework references of "Microsoft.NETCore.App" and "Microsoft.Windows.SDK.NET.Ref" - yet the default MSBuild pipeline still does implicitly add a framework reference to AspNetCore which I think is the cause of the problems.

**Comment by davidfowl** [comment]
Look at the assets file to see which project depends on the ASP.NET Core shared framework. Regardless, it seems this isn't a runtime issue but an issue with that package (if indeed it doesn't need the shared framework).

**Comment by a-teece** [comment]
@davidfowl None of my projects depend on ASP.NET framework, however looking at the nuspec for "Microsoft.Azure.SignalR.Management" I see one listed. Didn't even realise NuGet could include framework references.

Mystery solved, sadly not a great conclusion though.
